src/runtime/virtcontainers/endpoint.go

```diff
@@ -65,6 +65,9 @@ const (
 
 	// IPVlanEndpointType is ipvlan network interface.
 	IPVlanEndpointType EndpointType = "ipvlan"
+
+	// VfioEndpointType is VFIO device claimed as network interface by guest kernel
+	VfioEndpointType EndpointType = "vfio"
```

Comment:
The name and description here is quite confusing with `PhysicalEndpointType`.  `PhysicalEndpointType` was meant to represent an entire physical device or SRIOV interface that could be passed through VFIO. 
To clearly distinguish between the two, I suggest we should consider changing the naming here, maybe rename this to `DanEndPointType` or `DanVfioEndpointType`.
I also suggest to change the documentation comments to explicitly mention Direct attached networks. Without this, it will be a tall task for future developers to distinguish between the two endpoints.

Reply:
I will take a detailed look at this PR by tomorrow to provide any feedback.

Reply:
Thanks for looking at this PR.

If the name `VfioEndpointType` causing confusion, **I'm okay with `DanVfioEndpointType`** -- `DanEndPointType` looks like too generic. @justxuewei are you okay with `DanVfioEndpointType`? 

BTW: I saw `PhysicalEndpointType` unbinds VF device from its driver then rebind it to `vfio-pci` and cold-plug it to VM. When using ovn-kubernetes, the VF and VFIO devices are pre-allocated on the host, based on the network config on a workload(VM/Pod), a VF or VFIO device can be assigned to it accordingly -- a VFIO  device can be directly assigned to a VM workload, so no need to allocate and config a VF device then use PhysicalEndpoint to unbind/rebind it as VFIO, see https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/cni/helper_linux.go#L293-L301. 

One question out of curiosity: In the use case of PhysicalEndpoint, which CNI is typically utilized?

Reply:
@amshinde do you have further comments for this PR?

Reply:
@amshinde I added a doc string to clarify the difference from PhysicalEndpointType. I also renaming,  but since the endpoint type is inherently VFIO and DAN is merely the configuration method, renaming it to DanVfioEndpointType might introduce unnecessary confusion. Even if we renamed it, the value would still be 'vfio'. Therefore, I prefer to keep the current name. Does that work for you?"

Reply:
I don't think adding DAN is a good idea, since DAN stands for Directy Attachable Network. It just a new way of reading network config, and is not relevant for endpoints.

Reply:
@amshinde do you have any further feedback on this PR?

Reply:
> One question out of curiosity: In the use case of PhysicalEndpoint, which CNI is typically utilized?

We used this with sriov-device-network-plugin : https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin
The plugin creates a VF and places it in the network namespace of the sandbox.

Reply:
got delayed a bit, I'll take another look and provide my review by today.